src/ApplicationInsights.Kubernetes/ApplicationInsights.Kubernetes.csproj

```diff
@@ -3,7 +3,7 @@
     <Features>IOperation</Features>
   </PropertyGroup>
   <PropertyGroup>
-    <TargetFramework>net6.0</TargetFramework>
+    <TargetFramework>net8.0</TargetFramework>
```

Comment:
Bumping up the target to net8 would require all the apps upgrade to .NET 8, which is a major backward compatibility concerning.

It is a bit too early considering the rest of lifespan left for .NET 6:
![image](https://github.com/microsoft/ApplicationInsights-Kubernetes/assets/3674549/47d26f8f-d6b3-419f-a508-25770360b1d1)

It is mentioned that `It could have been done without upgrade`, I am wondering what alternatives are there? Do you mind try something without bumping up to .NET 8 (yet).

Reply:
How about just bumping up the package version of `Microsoft.Extensions.DependencyInjection` to 8.0.0? It is easier, and it supports `.net 6` and above.

Notice, technically, we could just reference `Microsoft.Extensions.DependencyInjection.Abstractions`, but that would lead to runtime exception like the one reported for .NET 6 applications when the referenced `Microsoft.Extensions.DependencyInjection` is older. To fix it, the user would have to reference of the higher version of it anyway.